lib/src/models/gateway/events/interaction.dart

```diff
@@ -4,10 +4,10 @@ import 'package:nyxx/src/models/interaction.dart';
 /// {@template interaction_create_event}
 /// Emitted when an interaction is received by the client.
 /// {@endtemplate}
-class InteractionCreateEvent<T extends Interaction<dynamic>> extends DispatchEvent {
+class InteractionCreateEvent<T extends Interaction<dynamic>> extends Event {
```

Comment:
There isn't a need for this change, a cleaner solution would simply be to have another type of interaction event emitted when using HTTP interactions.

For example, `InteractionCreateEvent` would only be emitted over the gateway, and `WebhookInteractionCreateEvent` (which would not extend `GatewayEvent`) would be emitted when using HTTP interactions.

Both events would simply contain an `interaction` field (and could both implement a common interface), but keeping `InteractionCreateEvent` as a `GatewayEvent` allows for tighter typing in a lot of places (e.g `Gateway.onEvent` and `EventMixin.onEvent`) and allows users to access the client associated with an interaction using `event.gateway.client`,